src/%IPM/Main.cls

```diff
@@ -2383,11 +2412,229 @@ ClassMethod UpdateLanguageExtensions(pVerbose As %Boolean = 0, pTestOnly As %Boo
+	Set resultSet = statement.%Execute(XMLDir, "ipm-*.xml")
+	If (resultSet.%SQLCODE < 0)  {
+		$$$ThrowSQLIfError(resultSet.%SQLCODE,resultSet.%Message)
+	}
+	Kill ipmLocalArray			// multi-dim array to store local ipm manifests; ipmLocalArray(<version>)=<fileName>
+	Kill menuList				// menu to let user choose which version of local ipm to install (if not in quiet mode and no version is specified)
+	While resultSet.%Next(.sc) {
+		$$$ThrowOnError(sc)
+		Set fileName = resultSet.%Get("Name")
+		If (resultSet.%Get("Type") = "F") {
+			// get the exact version string from filename
+			Set exactIPMVer = $Piece($Piece(fileName, "ipm-", 2, *), ".xml", 1, *-1)
+			Set ipmLocalArray(exactIPMVer) = fileName
+			Set menuList($Increment(menuList)) = exactIPMVer_" (local version)"
+		}
+	}
+	$$$ThrowOnError(sc)
+
+	If ('$DATA(menuList) && localOnly) {
+		$$$ThrowOnError($$$ERROR($$$GeneralError,"No ipm-<version>.xml installer file is found locally in directory: "__XMLDir))
+	}
+
+	// 2. Choose / propmpt user to choose which local/remote version of IPM to install
+	If (version '= "") {
+		Set targetVersion = version
+		If '$DATA(ipmLocalArray(version)) {
+			If localOnly {
+				$$$ThrowOnError($$$ERROR($$$GeneralError,"No ipm-"_version_".xml installer file is found locally in directory: "_XMLDir))
+			}
+			Write !, "No ipm-"_version_".xml installer file is found in "_XMLDir_" locally. Trying to download from the registry."
+			// need to get from remote registry server
+			Set useLocal = 0
+		}
+	} Else {
+		If quiet {
+			// Non-prompt mode + no version specified - either error out or install latest version from registry
+			If localOnly {
+				$$$ThrowOnError($$$ERROR($$$GeneralError,"To install local IPM in non-prompt mode, a specific version must be specified using the version modifier."))
+			} Else {
+				Write !, "No IPM version is specified and is in non-prompt mode. Install the latest version of IPM from the registry."
+				Set useLocal = 0
+				Set targetVersion = "latest"
+			}
+		} Else {
+			// prompt user with option to download the latest version from registry
+			Set localIPMCount = menuList
+			If 'localOnly {
+				// Add remote version options to menuList
+				Set server = ##class(%IPM.Repo.Remote.Definition).DeploymentServerOpen(1,,.sc)
+				$$$ThrowOnError(sc)
+				If $IsObject(server) {
+					Set latestVersion = server.GetPackageService().GetLatestIPMVersion()
+					Set menuList($Increment(menuList)) = "latest"_" (remote version - "_latestVersion_")"
+				}
+				kill server
+				kill packageService
+			}
+			Set menuValue = 1
+			Set menuResponse = ##class(%Library.Prompt).GetMenu("Which version of IPM to use?", .menuValue, .menuList,,128)
+			If (menuResponse '= 1) {
+				$$$ThrowOnError($$$ERROR($$$GeneralError,"Unsupported response. Operation cancelled."))
+			}
+			Set targetVersion = $Piece(menuList(menuValue), " ") // get rid of (local version) or (remote version) suffix
+			zw targetVersion
```

Comment:
Should not use zw. Should use a clear write statement if you want to write out the version selected.

Reply:
Ah this is debugging zw, now removed